flow/project.py

```diff
@@ -267,6 +268,25 @@ def _make_bundles(operations, size=None):
             break
 
 
+class _HooksDecorator:
```

Comment:
I would prefer the name `_HooksRegistery` to indicate its storage capacity before instantiation or some name that indicates its storage role. Alternatively we could add that information to a short developer docstring.